### PR TITLE
feat: [CDS-68759]: implement AWS SAM service changes

### PR DESCRIPTION
**Pull request by chetanvekariya-harness**
### Summary
- Show AWS SAM deployment type option
- Implement edit view component that does not have Artifact section
- Make tooltip ids unique for manifest add step tittles - https://harness.atlassian.net/browse/CDS-69569

#### Screenshots
Screenshot 2023-05-19 at 11 33 35 AM

Screenshot 2023-05-19 at 11 34 07 AM

Screenshot 2023-05-19 at 12 00 33 PM

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by chetanvekariya-harness** [comment]
retrigger jest

**Review by AshwinBhatkal-Harness — APPROVED** [review]

**Review by samarth-sharma31 — APPROVED** [review]

File: src/modules/70-pipeline/components/PipelineSteps/PipelineStepInterface.ts [hunk 1]
```diff
@@ -208,7 +208,8 @@ export enum StepType {
   Brakeman = 'Brakeman',
   Fossa = 'Fossa',
   Semgrep = 'Semgrep',
-  KubernetesAws = 'KubernetesAws'
+  KubernetesAws = 'KubernetesAws',
+  AwsSam = 'AWS_SAM'
```
Comment (YounixIO) [inline src/modules/70-pipeline/components/PipelineSteps/PipelineStepInterface.ts:212]: Should we be following CamelCasing here too?

File: src/modules/70-pipeline/utils/stageHelpers.ts [hunk 2]
```diff
@@ -76,7 +76,8 @@ export enum ServiceDeploymentType {
   SshWinRmAzure = 'SshWinRmAzure',
   Asg = 'Asg',
   GoogleCloudFunctions = 'GoogleCloudFunctions',
-  AwsLambda = 'AwsLambda'
+  AwsLambda = 'AwsLambda',
+  AwsSam = 'AWS_SAM'
```
Comment (YounixIO) [inline src/modules/70-pipeline/utils/stageHelpers.ts:80]: Should we be following CamelCasing here too?

Comment (chetanvekariya-harness) [reply src/modules/70-pipeline/utils/stageHelpers.ts:80]: I will discuss with BE and get it changed. Subsequently, I will also change it in one of my next PRs.

**Review by YounixIO — APPROVED** [review]

**Pull request merged**